Implement different file types for Entry.create_datum/Data.__new__

zngguvnf opened the pull request:
- wav
- flac
- ogg
- mat

zngguvnf committed:
implements option to save data as audio

zngguvnf committed:
makes fileformat implementation more flexible

zngguvnf committed:
implements option to save data as mat file

zngguvnf committed:
adjusts documentation

zngguvnf committed:
when fileformat='mat' return just the array

do not show .mat file header

zngguvnf committed:
reorganizes code

zngguvnf committed:
updates requirements

bastibe committed:
formalizes test suite

bastibe committed:
Merge branch 'master' into master

bastibe commented on jbof.py:
```diff
@@ -127,7 +138,15 @@ class Datum(numpy.ndarray):
         with metafile.open() as f:
             metadata = json.load(f)
-        data = numpy.load(metadata['_filename'])
+        extension = splitext(metadata['_filename'])[1]
```
please use `pathlib.Path.suffix`

zngguvnf replied:
DONE in 9605b75

bastibe commented on jbof.py:
```diff
@@ -105,15 +108,23 @@ def create_datum(self, name, data, metadata, fileformat='npy', samplerate=None):
+            numpy.save(datafilename, data)
+        elif fileformat in ['wav', 'flac', 'ogg']:
+            if samplerate:
+                soundfile.write(str(datafilename), data, int(samplerate))
```
please add `samplerate` to metadata.

zngguvnf replied:
DONE in 33835db

bastibe commented on jbof.py:
```diff
@@ -18,6 +18,9 @@
 import json
 from pathlib import Path
+import soundfile
+from os.path import splitext, basename
```
This should not be necessary. Please use `pathlib` methods instead.

bastibe commented on setup.py:
```diff
@@ -9,6 +9,6 @@
     license='GPL v3',
     py_modules=['runforrest'],
-    install_requires=['numpy'],
+    install_requires=['numpy', 'json', 'soundfile', 'scipy'],
```
No need to add `json`, as it is part of the standard lib.

zngguvnf replied:
DONE in 4178140

bastibe reviewed (changes requested):
Thank you for your contribution. Please change a few minor issues, and then we can merge.

Please also re-add your tests, and update the README with the completed task.

zngguvnf committed:
uses pathlib instead of os.path

zngguvnf committed:
adds samplerate to metadata

zngguvnf committed:
Merge branch 'master' of github.com:zngguvnf/jbof

zngguvnf committed:
Merge branch 'master' into master

zngguvnf committed:
removes json from dependencies

zngguvnf committed:
updates the README with the completed task

zngguvnf commented:
Thanks for the review. This was really helpful for me.
- I fixed the issues you pointed out 
- I updated the README.
- Right now, the test suite is not finished.

zngguvnf committed:
adds basic test for fileformats = ['wav', 'ogg', 'flac' , 'wav']

zngguvnf commented:
I added some test, they check:
- can the data be restored.
- Does the metadata include a samplerate, when using an audio format.
